### ISSUE TITLE
“scribles input" mode does not get a good alpha map

### ISSUE THREAD

**Issue by kkkmax** [MarcoForte/closed-form-matting#14]
I use the same command:

    Python closed_form_matting.py testdata/input_1.jpg testdata/scribles_1.jpg -o testdata/out_1.jp
![hello](https://user-images.githubusercontent.com/42377340/49868381-3ed44800-fe48-11e8-89fe-3d3e0011d953.png)
g

The background area is marked with black, and the foreground area is marked with white, but the resulting picture looks like a grayscale image. Do I ignore any details?

**Comment by wyshmily** [comment]
I got the same problem as @kkkmax 
![test1](https://user-images.githubusercontent.com/18300550/51825353-8db63e00-231f-11e9-9d4b-053f6800ecaf.jpg)  
![test1_scribble](https://user-images.githubusercontent.com/18300550/51825438-bb9b8280-231f-11e9-8d1a-a2e44426d03c.jpg)
![image](https://user-images.githubusercontent.com/18300550/51825573-05846880-2320-11e9-8f43-def4e0e7c964.png)



![image](https://user-images.githubusercontent.com/18300550/51825540-f6051f80-231f-11e9-95af-e38504f5c57d.png)

**Comment by shyyhs** [comment]
I got almost the same problem as this
![image](https://user-images.githubusercontent.com/20918444/52465564-620a4200-2bc2-11e9-86f0-876b5e51034e.png)
![image](https://user-images.githubusercontent.com/20918444/52465572-6898b980-2bc2-11e9-8e7d-24f7b5a7e2c4.png)
![image](https://user-images.githubusercontent.com/20918444/52465579-6f273100-2bc2-11e9-99cc-f6b457055e34.png)

Also, use the flower in the testdata but with slightly different scribbles will also cause this problem.

**Comment by merofeev** [comment]
This issue involves at least two problems:

1. Small changes to pixel values during scribbles image creation. For example here is the actual difference between input image and scribbles image used by @shyyhs:
```python
cv2.imwrite('vis.png', cv2.imread('scribbles.png') - cv2.imread('image.png'))
```
![vis](https://user-images.githubusercontent.com/6138700/52522396-f2fe1d80-2c95-11e9-9e82-f627e7e9caed.png)
As you can see a lot of pixels have changed values. This might happen due to using of lossy compression for scribbles image (e.g. jpeg was used by @kkkmax and @wyshmily). Another possible source of these issue is color profile conversion in image editor.

I have tried adding scribbles with Gimp to image provided by @shyyhs and saving image with scribbles to png and haven't experienced these issue.

2. Another issue is not perfectly white and black pixels. If you use brush tool with smooth edge to add scribbles, pixels around stroke edges will not be perfectly white/black. I believe that scribbles image from @wyshmily has such pixels. The current version of code can't play well with such pixels. I have submitted PR #15 to fix this.

To summarize:
1. Use lossless compression (e.g. png) for scribbles image
2. Use tool with hard edge (e.g. pencil) to draw scribbles
3. Consider using trimap input instead of scribbles. The trimap is much more robust way to supply user-input to matting.
